adapters/excon/lib/opentelemetry/adapters/excon/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Excon
+      module Middlewares
+        # Excon middleware for instrumentation
+        class TracerMiddleware < ::Excon::Middleware::Base
+          def request_call(datum) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+            begin
+              unless disable_span_reporting?(datum) || datum.key?(:otel_span)
+                tracer.start_span(
+                  datum[:path],
```

Comment:
This could lead to high cardinality span names. See: https://github.com/open-telemetry/opentelemetry-specification/pull/416/files. We probably need a quantization approach in the long run. We can discuss options in the short term. Note, this issue exists in Rack, Faraday, and RestClient as well. Are there any opinions on how we should handle this now (as in this PR).